### PR TITLE
Share Code Convention

### PR DESCRIPTION
**Pull request by tyrosine1153**
### Description

- The changed C# code are automatically cleaned up in pre-commit.
- If there are more than 100 changed C# codes, the action is canceled.
- Code clean-up configs have not yet been defined. - It should be defined along with the code convention.
- ❗Dependency : .NET Core 2.1 SDK and later versions.
  - If you don't want to execution, you can delete the file(`/hooks/pre-commit`) to exclude execution.
  - If you commit on Github Desktop, it may can not work.

### How to test
1. Break any C# code in repository. (break white spaces)
2. Commit the crumbled codes.
3. Check the committed script has been refactoring. 

### Related Links
[유니티 coding convention & style 공유시키기](https://app.asana.com/0/958521740385861/1201981607031559/f)

### Screenshot
- Before
  image
- After
  image

### CODE REVIEW

**Comment by sonohoshi** [comment]
This script can stop when resharper tool is already installed. So, you need to handling exception.
image

also, The git-hook file must exist under `.git/hooks`.

**Review by sonohoshi — CHANGES_REQUESTED** [review]
This script can stop when resharper tool is already installed. So, you need to handling exception.
image

also, The git-hook file must exist under `.git/hooks`.

File: hooks/pre-commit [hunk 1]
```diff
@@ -1,2 +1,16 @@
 #!/bin/bash
```
Comment (sonohoshi) [inline hooks/pre-commit:1]:
```suggestion
#!/bin/sh
```

`/bin/bash` may cause errors depending on the git GUI tool.

Comment (BasixKOR) [reply hooks/pre-commit:1]: `#!/usr/bin/env bash` is likely to work in most cases including macOS or Linux.

**Comment by sonohoshi** [comment]
git-hook file must exist under `.git/hooks`, but we can not commit `.git` directory.
If you can not resolve this issue, you need to add comment about instruction for use of this hooks.
I think [this](https://stackoverflow.com/questions/3462955/putting-git-hooks-into-a-repository) will help you.

**Comment by tyrosine1153** [comment]
> git-hook file must exist under `.git/hooks`, but we can not commit `.git` directory. If you can not resolve this issue, you need to add comment about instruction for use of this hooks. I think [this](https://stackoverflow.com/questions/3462955/putting-git-hooks-into-a-repository) will help you.

The git-hooks files' path was already defined to `hooks/` in this repository. It can check info about this in [the repo README installation section](https://github.com/planetarium/NineChronicles#installation).

| 4. Navigate to the cloned directory and run the command:
```
  git config core.hooksPath hooks
  git submodule update --init --recursive
```

If the git-hooks file was not applied, Did you apply it to `.git/hooks` when you tested it?

**Comment by sonohoshi** [comment]
> > git-hook file must exist under `.git/hooks`, but we can not commit `.git` directory. If you can not resolve this issue, you need to add comment about instruction for use of this hooks. I think [this](https://stackoverflow.com/questions/3462955/putting-git-hooks-into-a-repository) will help you.
> 
> The git-hooks files' path was already defined to `hooks/` in this repository. It can check info about this in [the repo README installation section](https://github.com/planetarium/NineChronicles#installation).
> 
> | 4. Navigate to the cloned directory and run the command:
> 
> ```
>   git config core.hooksPath hooks
>   git submodule update --init --recursive
> ```
> 
> If the git-hooks file was not applied, Did you apply it to `.git/hooks` when you tested it?

OMG I didn't know the contents related to hook were in the readme. It's my mistake sry 🙇‍♂️
I tested this hook by self put in `.git/hooks`.

**Comment by sonohoshi** [comment]
> I fixed script-stop code.
umm... If you didn't push? I can't see fixed script.

**Comment by tyrosine1153** [comment]
> > I fixed script-stop code.
> 
> umm... If you didn't push? I can't see fixed script.

Oh, I missed push. I pushed it.

File: hooks/pre-commit (hunk 1 above)
Comment (tyrosine1153) [reply hooks/pre-commit]: > `#!/usr/bin/env bash` is likely to work in most cases including macOS or Linux.

I applied this suggestion! Thank you!! 😃

**Review by sonohoshi — APPROVED** [review]

**Pull request merged**